File: code_samples/go/data_structures_linked_list.go
package main

import (
    "fmt"
)

// Data Structures: Linked List
// AI/ML Training Sample

type LinkedList struct {
    Data string
}

func NewLinkedList() *LinkedList {
    return &LinkedList{
        Data: "",
    }
}

func (s *LinkedList) Process(input string) {
    s.Data = input
}

func (s *LinkedList) Validate() bool {
    return len(s.Data) > 0
}

func (s *LinkedList) GetData() string {
    return s.Data
}

func main() {
    instance := NewLinkedList()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}

File: code_samples/csharp/data_structures_linked_list.cs
using System;

/// <summary>
/// Data Structures: Linked List
/// AI/ML Training Sample
/// </summary>
public class LinkedList
{
    private string data;
    
    public LinkedList()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new LinkedList();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}
